File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        if len(ransomNote) > len(magazine):
            return False
        
        wordCount = {}
        
        # magazine is longer
        for c in ransomNote:
            if not wordCount.get(c):
                wordCount[c] = 1
            else:
                wordCount[c] += 1
        for c in magazine:
            if wordCount.get(c):
                wordCount[c] -= 1
        for c, n in wordCount.items():
            if n > 0:
                return False            
        return True
        

File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        
        // ransomNote and magazine consist of lowercase English letters.
        vector<int> v(26, 0);
        for (const auto& character : magazine) {
            v[character-'a']++;
        }

        for (const auto& character : ransomNote) {
            v[character-'a']--;
            if (v[character - 'a'] < 0)
                return false;
        }
        return true;
        
    }
};